test/system/065-cp.bats

```diff
@@ -7,6 +7,312 @@
+
+    srcdir=$PODMAN_TMPDIR/cp-test-file-host-to-ctr
+    mkdir -p $srcdir
+    # Create multiple files which prevents some (expensive) clean ups via
+    # podman-exec.
+    echo "This file is on the host" > $srcdir/hostfile
+    echo "This file is on the host (1)" > $srcdir/hostfile1
+    echo "This file is on the host (2)" > $srcdir/hostfile2
+
+    run_podman run -d --name cpcontainer --workdir=/srv $IMAGE sleep infinity
+
+    # Copy to root.
+    run_podman cp $srcdir/hostfile cpcontainer:/
+    run_podman exec cpcontainer cat /hostfile
+    is "$output" "This file is on the host"
+
+    run_podman cp $srcdir/hostfile cpcontainer:/anotherbase.txt
+    run_podman exec cpcontainer cat /anotherbase.txt
+    is "$output" "This file is on the host"
+
+    # Copy to a non-root directory (varied input syntax).
+    run_podman cp $srcdir/hostfile cpcontainer:/tmp
+    run_podman exec cpcontainer cat /tmp/hostfile
+    is "$output" "This file is on the host"
+
+    run_podman cp $srcdir/hostfile1 cpcontainer:/tmp/
+    run_podman exec cpcontainer cat /tmp/hostfile1
+    is "$output" "This file is on the host (1)"
+    run_podman exec cpcontainer rm /tmp/hostfile
+
+    run_podman cp $srcdir/hostfile2 cpcontainer:/tmp/.
+    run_podman exec cpcontainer cat /tmp/hostfile2
+    is "$output" "This file is on the host (2)"
+
+    run_podman cp $srcdir/hostfile cpcontainer:/tmp/anotherbase.txt
+    run_podman exec cpcontainer cat /tmp/anotherbase.txt
+    is "$output" "This file is on the host"
+
+    # Copy to a relative path -> must resolve to the container's work dir
+    # (i.e., /srv).
+    run_podman exec cpcontainer mkdir /srv/subdir
+    run_podman cp $srcdir/hostfile cpcontainer:subdir
+    run_podman exec cpcontainer cat /srv/subdir/hostfile
+    is "$output" "This file is on the host"
+
+    run_podman cp $srcdir/hostfile cpcontainer:anotherbase.txt
+    run_podman exec cpcontainer cat /srv/anotherbase.txt
+    is "$output" "This file is on the host"
+
+    # Host path does not exist.
+    run_podman 125 cp $srcdir/IdoNotExist cpcontainer:/tmp
+    is "$output" 'Error: ".*/IdoNotExist" could not be found on the host'
+
+    # Container path does not exist.  Notice that the error message shows how
+    # the specified container is resolved.
+    run_podman 125 cp $srcdir/hostfile cpcontainer:/IdoNotExist/
+    is "$output" 'Error: "/IdoNotExist/" could not be found on container.*(resolved to .*/IdoNotExist.*'
+
+    run_podman rm -f cpcontainer
+}
+
+
+@test "podman cp --extract=true tar archive to container" {
+    skip_if_remote "podman-remote does not yet handle cp"
+
+    # Create tempfile with random name and content
+    srcdir=$PODMAN_TMPDIR/cp-test-stdin
+    mkdir -p $srcdir
+    rand_filename=$(random_string 20)
+    rand_content=$(random_string 50)
+    echo $rand_content > $srcdir/$rand_filename
+    chmod 644 $srcdir/$rand_filename
+
+    # Now tar it up!
+    tar_file=$PODMAN_TMPDIR/archive.tar.gz
+    tar -zvcf $tar_file $srcdir
```

Comment:
Absolute paths in tarballs make me nervous. Would you consider using `-C` instead, with corresponding changes below (`s/$srcdir/cp-test-stdin/`, possibly with a refactoring of the hardcoded `cp-test-stdin`, or even better, a randomly-generated dir name? Random is IMHO better than misleading, and the 'stdin' here is misleading):
```bash
    tar -C $PODMAN_TMPDIR -zvcf $tar_file .
```

Reply:
done